### PR TITLE
bfg: add page

### PR DESCRIPTION
**Pull request by mnahinkhan**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

**Comment by CleanMachine1** [comment]
Thank you for your contributions @mnahinkhan 

Your work is very much appreciated.
We need your opinion with things and speed is wanted, however not required.
Your PR maybe merged in a few days normally.
Please stick with us and give your opinions as we suggest changes

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:16]: We normally don't reference other commands. We may add something to one of the other command, like these changes are only commited, not pushed to origin

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:12]:
```suggestion
- Replace all text listed in text.txt wherever it can be found in the repository's history:

`bfg --replace-text {{text.txt}}`
```

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:8]:
```suggestion
`bfg --delete-files {{file_with_sensitive_data}}`
```

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:12]:
```suggestion
- Remove all text mentioned in text.txt wherever it can be found in the repository's history:

`bfg --replace-text {{text.txt}}`
```

File: pages/common/bfg.md
Comment (CleanMachine1) [reply pages/common/bfg.md:16]: @bl-ue what's your opinion on this

**Comment by CleanMachine1** [comment]
@bl-ue can I approve this workflow. It seems ok

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:3]:
```suggestion
> Remove large files or passwords from Git history like git-filter-branch.
```

As soon as the workflow is approved, the bot will flag this

File: pages/common/bfg.md
Comment (CleanMachine1) [inline pages/common/bfg.md:16]:
```suggestion
`git push --force`

```
Need a new line at the end \n

**Comment by bl-ue** [comment]
>@bl-ue can I approve this workflow. It seems ok

@CleanMachine1 always approve the workflows, unless it's a spam PR 👍🏻

File: pages/common/bfg.md
Comment (bl-ue) [reply pages/common/bfg.md:12]: I'd rather something a bit more generic, e.g.
```suggestion
- Remove all text mentioned in the specified file wherever it can be found in the repository's history:

`bfg --replace-text {{path/to/file.txt}}`
```

File: pages/common/bfg.md
Comment (CleanMachine1) [reply pages/common/bfg.md:16]: Also this needs to not be Github, just origin since gitlab and bitbucket do exist

Comment (bl-ue) [reply pages/common/bfg.md:16]: There's a few problems with this example:
1. As @CleanMachine1 points out, it's a different command other than the one that we're documenting, but that's of the least concern
2. The command assumes that the repository is connected to a remote, which it may, in fact, not be
3. GitHub isn't the only remote repository service ;)

I suggest moving this information to a new line in the page description, something like
```
> Note: you will need to force-push if your repository is connected to a remote.
```

Comment (mnahinkhan) [reply pages/common/bfg.md:16]: Thank you so much for your detailed feedback! I'll make these changes within a day or two when I get some time 😊

File: pages/common/bfg.md
Comment (bl-ue) [inline pages/common/bfg.md:7]: "And" makes it sound as though it's doing an action, but it's really just an omission.
```suggestion
- Remove a file with sensitive data but leave the latest commit untouched:
```

Comment (mnahinkhan) [reply pages/common/bfg.md]: I agree!

**Review by bl-ue — APPROVED** [review]
Looks good to me! I'm surprised this page hasn't been added before 🤔

Thank you @mnahinkhan! 🚀

**Comment by mnahinkhan** [comment]
Thank you for your involvement and guidance I have really enjoyed this process!

**Review by CleanMachine1 — APPROVED** [review]
```
                                                                       xmHTTTTT%ms.
                                                                    z?!!!!!!!!!!!!!!?m
                                                                  z!!!!!!!!!!!!!!!!!!!!%
                                                               eHT!!!!!!!!!!!!!!!!!!!!!!!L
                                                              M!!!!!!!!!!!!!!!!!!!!!!!!!!!>
                                                           z!!!!!!!!!!XH!!!!!!!!!!!!!!!!!!X
                                                           "$$F*tX!!W?!!!!!!!!!!!!!!!!!!!!!
                          good first PR                   >     M!!!   4$$NX!!!!!!!!!!!!!t
                                                           tmem?!!!!?    ""   "X!!!!!!!!!!F
                                                      um@T!!!!!!!!!!!!s.      M!!!!!!!!!!F
                                                   .#!!!!!!!!!!!!!!!XX!!!!?mM!!!!!!!!!!t~
                                                  M!!!@!!!!X!!!!!!!!!!*U!!!!!!!!!!!!!!@
                                                 M!!t%!!!W?!!!XX!!!!!!!!!!!!!!!!!!!!X"
                                                :!!t?!!!@!!!!W?!!!!XWWUX!!!!!!!!!!!t
                                                4!!$!!!M!!!!8!!!!!@$$$$$$NX!!!!!!!!-
                                                 *P*!!!$!!!!E!!!!9$$$$$$$$%!!!!!!!K
                                                    "H*"X!!X&!!!!R**$$$*#!!!!!!!!!>
                                                        'TT!?W!!9!!!!!!!!!!!!!!!!M
                                                        '!!!!!!!!!!!!!!!!!!!!!!!!F
                                                        '!!!!!!!!!!!!!!!!!!!!!!!!>
                                                        '!!!!!!!!!!!!!!!!!!!!!!!M
                                                        J!!!!!!!!!!!!!!!!!!!!!!!F K!%n.
         @!!!!!??m.                                  x?F'X!!!!!!!!!!!!!!!!!!!!HP X!!!!!!?m.
Z?L      '%!!!!!!!!!?s                            .@!\~ MB!!!!!!!!!!!!!!!!!U#!F X!!!!!!!!X#!%.
E!!N!k     't!!!!!!!!!?:                       zTX?!t~ M!t!!!!!!!!!!!!!!UM!!!F 4!!!!!!!!t%!!!!?.
!!!!!!hzh.   "X!!!!!!!!!>                  .+?!!3?!X  Z!!!B!!!!!!!!!!UM!!!!!" 4!!!!!!!!t?!!!!!!!h
?!!!!!!!!!*!?L %!!!!!!!!?               .+?!!!!3!!\  P!!!!?X!!!!!!U#!!!!!!X" 4!!!!!!!!\%!!!!!!!!!?
'X!!!!!!!!!!!!?TTTT*U!!!!k            z?!!!!!!t!!!- J!!!!!!9!!X@T!!!!!!!!X~ d!!!!!!!!!%!!!!!!!!!!!!L
 4!!!!!!!!!!!!!!!!!!!!!!!M          'W!!!!!!!X%!!P  %!!!!!!!T!!!!!!!!!!!X~ J!!!!!!!!!P!!!!!!!!!!!!!!\
  5!!!!!!!!!!!!!!!!!!!!!!!?m.       .@Ti!!!!!Z!!t  d!!!!!!!!!!!!!!!!!!!X-.JUUUUX!!!!J!!!!!!!!!!!!!!!!!
   %!!!!!!!!!!!!!!!!!!!!!!!!!!!TnzT!!!!!#/!!?!!X"  ^"=4UU!!!!!!!!!!U@T!!!!!!!!!!!!Th2!!!!!!!!!!!!!!!!!!
    ^t!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!?K!K!!f               `""#X!!!!!!!!!!!!!!!!!?t!!!!!!!!!!!!!!!!(>
       "U!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!$!!F                      "tX!!!!!!!!!!!!!!!!b!!!!!!!!!!!!!!!(>
          '"*tUUX!X!!!!!!!!!!!!!!!!!!!!!!!!$!Z                          ^4!!!!!!!!!!!!!!!N!!!!!!!!!!!!!!!!
                 %!!!!!!!!!!!!!!!!!!!!!!!!X!X                              %W@@WX!!!!!!!!!N!!!!!!!!!!!!!!!
                  "X!!!!!!!!!!!!!!!!!!!!!@!!*        ..    ..  :m.. ETThmuM!!!!!!!!!!!!!!!!@m@*TTTT?!!!W!!
                    %!!!!!!!!!!!!!!!!!!W?!!X         M!!!TT?!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!9UU!!!!!!!!!M!f
                     't!!!!!!!!!!!!!!!P!!!!X          5!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!?NX!!!!!!L
                       "W!!!!!!!!!!!X#!!!!!R           "X!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!R!!!!!t
                         ^*X!!!!!!!t%!!!!!h              %X!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!>
                             "*U!!M!!!!!!X~ :?!!!T!+s...   *X!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                                 :?!!!!!!> :?!!!!!!!!!!!!!!!!?tX!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!>
                                 %!!!!!!F .%!!!!!!!!!!!!!!!!!!!!!#4U!!!!!!!!!!!!U!!!!!!!!!!!!!!!!!!!!!!!!~
                                K!!!!!!Z  K!!!!!!!!!!!!!!!!!!!!!!!  F!!!!!?!!?X!!!!!!!!!!!!!!!!!!!!!!!!!Z
                               X!!!!!!t  H!!!!!!!!!!!!!!!!!!!!!!!!> !!!!!!!!!!!W!!!!!!!!!!!!!!!!!!!!!!!t
                               %!!!!!!F :!!!!!!!!!!!!!!!!!!!!!!!!!> !!!!!!!!!!!!#X!!!!!!!!!!!!!!!!!!!!X
                              '!!!!!!X  K!!!!!!!!!!!!!!!!!!!!!!!!!> K!!!!!!!!!!!!!?W!!!!!!!!!!!!!!!!X"
```

**Comment by CleanMachine1** [comment]
Hope you enjoy Dr.Z :)

**Comment by mnahinkhan** [comment]
Haha thank you again for your time and this amazing project. Feel free to close ofc :)

**Comment by CleanMachine1** [comment]
We wouldn't close unless it is pointless or spam.
Do you plan on helping any more, we always need help and welcome more pages/improvements
@navarroaxel could you review so we can merge without delay

**Comment by mnahinkhan** [comment]
> We wouldn't close unless it is pointless or spam.
> Do you plan on helping any more, we always need help and welcome more pages/improvements
> @navarroaxel could you review so we can merge without delay

I see, I just assumed PR's are meant to be closed after approval; sorry I'm new to this.

As for helping, there are often times when I type 'tldr <something>' in the terminal only to find it doesn't exist. In those times I'm usually in a rush, so I quickly look up the info in the man page or the internet. This happened enough times that I decided to contribute this time, but by now I've forgotten the tools for which the pages didn't exist 😅. This is obviously not the place to suggest this, but if TLDR could give me like a list of recent calls that did not have a page I would be more than happy to write more pages!

But I'll keep an eye out for more pages that are missing and contribute :)

**Comment by mnahinkhan** [comment]
Ah actually on second thought I could check my shell history, but sometimes those get lost here and there as I keep tinkering with new environments.

**Comment by mnahinkhan** [comment]
Also if there are ways for me to contribute without necessarily having to add more pages, I am happy to contribute some time. I'll read up on the contribution guidelines and see how I can help out

**Comment by CleanMachine1** [comment]
I started with tldr because I found it in my explore. I just starred it and moved on, then I found a command #5923 and I was hooked with helping!

**Review by navarroaxel — APPROVED** [review]

**Comment by bl-ue** [comment]
Same here. I started by cleaning up via typo-fixing in #5004 (almost 1000 PRs ago! 👀), and the standards for adding collaborators interested me a lot.  Having reviewed [675 PRs](https://github.com/tldr-pages/tldr/pulls?q=is%3Apr+reviewed-by%3Abl-ue+) since then, I've come a long way! :)

**Pull request merged**
